File: SumRootToLeafNumbers/SumRootToLeafNumbers.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution(object):
    def sumNumbers(self, root):
        """
        :type root: TreeNode
        :rtype: int
        """
        if root is None:
            return 0
        return search(root, 0)

def search(root, val):
    if root.left is None and root.right is None:
        return 10 * val + root.val
    tmp = 0
    if root.left is not None:
        tmp += search(root.left, 10 * val + root.val)
    if root.right is not None:
        tmp += search(root.right, 10 * val + root.val)
    return tmp

File: SumRootToLeafNumbers/SumRootToLeafNumbers.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
public:
    int sumNumbers(TreeNode* root) {
        if(root == nullptr)
        	return 0;
        vector<int> subSums;
        getSum(root, root->val, subSums);
        int result = 0;
        for(int i = 0; i < subSums.size(); i++) {
        	result += subSums[i];
        }
        return result;
    }
private:
	void getSum(TreeNode* node, int subSum, vector<int> &sumSums) {
		if(node->left == nullptr && node->right == nullptr) {
			sumSums.push_back(subSum);
			return;
		}
		if(node->left != nullptr)
			getSum(node->left, 10*subSum + node->left->val, sumSums);
		if(node->right != nullptr)
			getSum(node->right, 10*subSum + node->right->val, sumSums);
		return;
	}
};